src/machine/machinectl.c

```diff
@@ -117,9 +118,56 @@ static int compare_machine_info(const void *a, const void *b) {
         return strcmp(x->name, y->name);
 }
 
+static int get_os_name(sd_bus *bus, const char *name, char **out)
+{
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+        const char *k, *v;
+        int r;
+
+        assert(bus);
+        assert(name);
+        assert(out);
+
+        r = sd_bus_call_method(bus,
+                "org.freedesktop.machine1",
+                "/org/freedesktop/machine1",
+                "org.freedesktop.machine1.Manager",
+                "GetMachineOSRelease",
+                NULL, &reply, "s", name);
+
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_enter_container(reply, 'a', "{ss}");
+
+        if (r < 0)
+                return bus_log_parse_error(r);
+
+        *out = NULL;
```

Comment:
Please don't do that. The function should set the output arguments only iff it succeeds.